app/views/layouts/base.html.erb

```diff
@@ -15,7 +15,19 @@
   <meta content="noindex,nofollow" name="robots" />
   <% end %>
 
-  <title><%= yield(:title).presence || (raise 'Missing title') %> | <%= APP_NAME %></title>
+  <%
+    title = begin
```

Comment:
Maybe a little too "magical" / "clever", but maybe we could adapt [the `title` application helper](https://github.com/18F/identity-idp/blob/d5f82fb36ac9bf9ab5ec13889a2067851b857103/app/helpers/application_helper.rb#L4-L6) to contain all the logic here if it's called without an argument?

Reply:
doing that would detect if we explicitly set a nil title for a page, but it wouldn't detect the absence of a title for the page right?

Reply:
I don't know if I'm following the concern. This is what I had in mind:

```rb
module ApplicationHelper
  class MissingTitleError < StandardError; end

  def title(title = nil)
    if title.present?
      content_for(:title) { title }
    else
      content_for(:title).presence || (raise MissingTitleError)
    end
  rescue MissingTitleError => error
    if IdentityConfig.store.raise_on_missing_title
      raise error
    else
      NewRelic::Agent.notice_error(error)
      ''
    end
  end

  # ...
end
```

Reply:
ahhhh yes now I see, I will make that change

Reply:
The part that confused me about your suggestion was the combined setter + getter aspect, so I split those out and made a new method in 4a2db79c51c8dc191764e6e082081b21bf768b46, PTAL!

Reply:
Yeah, was kinda wondering if it'd be "better" or even possible to do `def title=(title)` to separate the methods while sharing the same name, since conceptually they overlap. But even if it did work, it'd touch _a lot_ of files, so not necessarily something I'd propose to take on here.

Reply:
We'd have to make the templates call `self.title = ` because otherwise Ruby will parse it as a local variable assignment, but yes that would cause a lot of diff noise that I'd like to avoid for now

Reply:
I was interested 😄 #9529